components/event-bus/internal/knative/subscription/controller/eventactivation/reconcile.go

```diff
@@ -3,114 +3,131 @@ package eventactivation
+	"github.com/kyma-project/kyma/components/event-bus/internal/knative/util"
 )
 
 const (
-	finalizerName = "eventactivation.finalizers.kyma-project.io"
+	finalizerName                   = "eventactivation.finalizers.kyma-project.io"
+	eventactivationreconciled       = "EventactivationReconciled"
+	eventactivationreconciledfailed = "EventactivationReconcileFailed"
 )
 
-type reconciler struct {
-	client   client.Client
-	recorder record.EventRecorder
-	time     util.CurrentTime
-}
+//Reconciler EventActivation reconciler
+type Reconciler struct {
+	// wrapper for core controller components (clients, logger, ...)
+	*reconciler.Base
 
-// Verify the struct implements reconcile.Reconciler
-var _ reconcile.Reconciler = &reconciler{}
+	// listers index properties about resources
+	eventActivationLister applicationconnectorlistersv1alpha1.EventActivationLister
 
-func (r *reconciler) InjectClient(c client.Client) error {
-	r.client = c
-	return nil
+	// clients allow interactions with API objects
+	applicationconnectorClient applicationconnectorclientv1alpha1.ApplicationconnectorV1alpha1Interface
+	kymaEventingClient         eventingclientv1alpha1.EventingV1alpha1Interface
+
+	time util.CurrentTime
 }
 
 // Reconcile reconciles a EventActivation object
-func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	log.Info("Reconcile: ", "request", request)
-
-	ctx := context.TODO()
-	ea := &eventingv1alpha1.EventActivation{}
-	err := r.client.Get(ctx, request.NamespacedName, ea)
-
-	// The EventActivation may have been deleted since it was added to the workqueue. If so, there is
-	// nothing to be done.
-	if errors.IsNotFound(err) {
-		log.Info("Could not find EventActivation: ", "err", err)
-		return reconcile.Result{}, nil
-	}
-
-	// Any other error should be retrieved in another reconciliation. ???
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	ea, err := eventActivationByKey(key, r.eventActivationLister)
 	if err != nil {
-		log.Error(err, "Unable to Get EventActivation object")
-		return reconcile.Result{}, err
+		return err
 	}
 
-	log.Info("Reconciling Event Activation", "UID", string(ea.ObjectMeta.UID))
-
 	// Modify a copy, not the original.
 	ea = ea.DeepCopy()
 
 	// Reconcile this copy of the EventActivation and then write back any status
 	// updates regardless of whether the reconcile error out.
-	requeue, reconcileErr := r.reconcile(ctx, ea)
+	reconcileErr := r.reconcile(ctx, ea)
 	if reconcileErr != nil {
-		log.Error(reconcileErr, "Reconciling EventActivation")
-		r.recorder.Eventf(ea, corev1.EventTypeWarning, "EventactivationReconcileFailed", "Eventactivation reconciliation failed: %v", reconcileErr)
+		r.Recorder.Eventf(ea, corev1.EventTypeWarning, eventactivationreconciledfailed,
+			"Eventactivation reconciliation failed: %v", reconcileErr)
```

Comment:
I think a log entry would be more sufficient, otherwise the user has to track failures in multiple places. Events should be reserved for useful state changes IMO.
I'd keep the call to `log.Error` and remove the event.

Reply:
I agree that we need to add a log entry, looks like missed in the refactor. However, I think from the reconciler lifecycle POV, an Event in case of failure still usefull, what do you think?

Reply:
@antoineco 🔼

Reply:
I can live with that, as long as the error message is useful 👍